### PR TITLE
Azure integration tests

### PR DESCRIPTION
**Pull request by gonzalogarciajaubert**
## Description
Implement deploy and nuke tests for azure and make the integration suite run on the deployed Azure stack.

## Changes
1. Add Azure Deploy Integration Test
1. Add Azure Nuke Integration Test
1. Fix bug creating infrastructure on azure, k8s, and local providers (related to this [PR](https://github.com/boostercloud/booster/pull/957))
1. Add has-tostringtag package to avoid  `eslint` errors
1. Refactor run-command.ts (move it to framework-common-helpers)

## Checks
- [x] Project Builds
- [x] Project passes tests and checks
- [x] Updated documentation accordingly
 
## Additional information

This change increases the `Nuke` test timeout to avoid issues with Azure.

Not done:
1. Update Github workflow. Pending improvement `Nuke` performance

### CODE REVIEW

**Review by javiertoledo — APPROVED** [review]
Looks great! Just left a few comments

File: packages/framework-integration-tests/integration/provider-specific/azure/deployment/setup.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  await overrideWithBoosterLocalDependencies(sandboxedProject)
+
+  const servicePrincipalName = applicationName() + '-sp'
+  await AzureTestHelper.preBuild(servicePrincipalName)
+
+  console.log('Deploying sandbox project')
+  await deploy(sandboxedProject, 'azure')
+  console.log('Waiting 30 seconds after deployment to let the ARM finish its initialization...')
+  await sleep(30000)
+  console.log('...sleep finished. Let the tests begin 🔥!')
```
[the remark this answers is not in the document]
Comment (gonzalogarciajaubert) [reply packages/framework-integration-tests/integration/provider-specific/azure/deployment/setup.ts:23]: cool, let me remove it and try it

File: packages/framework-integration-tests/integration/provider-specific/azure/deployment/setup.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  await setEnv()
+  const configuredAssets = ['assets', 'assetFile.txt']
+  const sandboxedProject = createSandboxProject(sandboxPathFor('deploy'), configuredAssets)
+
+  await overrideWithBoosterLocalDependencies(sandboxedProject)
+
+  AzureTestHelper.ensureAzureConfiguration()
+  console.log('Deploying sandbox project...')
+  await deploy(sandboxedProject, 'azure')
+  console.log('...sleep finished. Let the tests begin 🔥!')
```
Comment (alvaroloes) [inline packages/framework-integration-tests/integration/provider-specific/azure/deployment/setup.ts:20]:
```suggestion
  console.log('Finished! Let the tests begin 🔥!')
```

(as you removed the sleep 😉 )

**Review by alvaroloes — APPROVED** [review]
🎉 🎉 🎉 🎉  Thanks a lot @gonzalogarciajaubert 
I left just a suggestion for a log message

File: packages/framework-integration-tests/integration/provider-specific/azure/deployment/setup.ts (hunk 2 above)
Comment (gonzalogarciajaubert) [reply packages/framework-integration-tests/integration/provider-specific/azure/deployment/setup.ts:20]: Right! thanks!

**Pull request merged**